addrmgr/addrmanager.go

```diff
@@ -823,6 +825,8 @@ func (a *AddrManager) Attempt(addr *wire.NetAddress) {
 	if ka == nil {
 		return
 	}
+	ka.mtx.Lock()
+	defer ka.mtx.Unlock()
```

Comment:
Please avoid defer for trivial cases since it has non-trivial overhead.  e.g.

```Go
// set last tried time to now
ka.mtx.Lock()
ka.attempts++
ka.lastattempt = time.Now()
ka.mtx.Unlock()
```

Reply:
Thanks davecgh, this was a miss :) will fix it